File: packages/awatv_core/lib/src/models/episode.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'episode.freezed.dart';
part 'episode.g.dart';

/// One episode inside a [SeriesItem].
@freezed
class Episode with _$Episode {
  const factory Episode({
    required String id,
    required String seriesId,
    required int season,
    required int number,
    required String title,
    required String streamUrl,
    String? plot,
    int? durationMin,
    String? posterUrl,
    String? containerExt,
  }) = _Episode;

  factory Episode.fromJson(Map<String, dynamic> json) =>
      _$EpisodeFromJson(json);
}

File: apps/apple_tv/Sources/AWAtv/Models/Episode.swift
import Foundation

/// One episode inside a `SeriesItem`.
struct Episode: Identifiable, Codable, Hashable, Sendable {
    let id: String
    let seriesId: String
    let season: Int
    let number: Int
    let title: String
    let streamUrl: String
    let plot: String?
    let durationMin: Int?
    let posterUrl: String?
    let containerExt: String?

    init(
        id: String,
        seriesId: String,
        season: Int,
        number: Int,
        title: String,
        streamUrl: String,
        plot: String? = nil,
        durationMin: Int? = nil,
        posterUrl: String? = nil,
        containerExt: String? = nil
    ) {
        self.id = id
        self.seriesId = seriesId
        self.season = season
        self.number = number
        self.title = title
        self.streamUrl = streamUrl
        self.plot = plot
        self.durationMin = durationMin
        self.posterUrl = posterUrl
        self.containerExt = containerExt
    }

    enum CodingKeys: String, CodingKey {
        case id
        case seriesId = "series_id"
        case season
        case number
        case title
        case streamUrl = "stream_url"
        case plot
        case durationMin = "duration_min"
        case posterUrl = "poster_url"
        case containerExt = "container_ext"
    }
}
